libpod/runtime_pod_linux.go

```diff
@@ -92,40 +92,50 @@ func (r *Runtime) NewPod(options ...PodCreateOption) (*Pod, error) {
 		return nil, errors.Wrapf(err, "error adding pod to state")
 	}
 
+	if pod.HasPauseContainer() && pod.SharesNamespaces() {
```

Comment:
Hm. I feel like we should always create a pause container here, even if the pod doesn't share namespaces...

Maybe we should make this an error - it doesn't make sense to have a pause container if you're not sharing namespaces, no?